code/modules/interface/multiContext/context_actions.dm

```diff
@@ -480,6 +489,16 @@
 					user.show_text("You wrench [target]'s bolts.", "blue")
 					playsound(target, 'sound/items/Ratchet.ogg', 50, 1)
 					return ..()
+				if(istype(I, /obj/item/tool/omnitool))
+					var/obj/item/tool/omnitool/omni = I
+					if (!(OMNI_MODE_WRENCHING in omni.modes))
+						continue
+					omni.change_mode(OMNI_MODE_WRENCHING, user, /obj/item/wrench)
+					user.show_text("You flip [omni] to wrenching mode...", "blue")
+					sleep(OMNI_TOOL_WAIT_TIME)
+					user.show_text("...then wrench [target]'s bolts.", "blue")
+					playsound(target, 'sound/items/Crowbar.ogg', 50, 1)
+					return ..()
```

Comment:
could move this to the parent and define an `omni_mode` var on each child?

Reply:
restructured to include this in [3eafef4](https://github.com/goonstation/goonstation/pull/12356/commits/3eafef46ef91b737d02a9a0c1a508f6f71f4ab69)